File: hw/maisincho/u3/HW23Observer/ObserverPy/model/IBM.py
from model.Stock import Stock
class IBM(Stock):
    def __init__(self, symbol, price):
        super().__init__()
        self._symbol = symbol
        self._price = price

    def get_symbol(self):
        return self._symbol

    def set_symbol(self, symbol):
        self._symbol = symbol
        self.notify_observers(symbol)

    def get_price(self):
        return self._price

    def set_price(self, price):
        self._price = price
        self.notify_observers(price)

File: ws/jacome/U3/WS35Observer/HW23Observer/src/main/java/ec/edu/espe/hw23observer/model/IBM.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */

package ec.edu.espe.hw23observer.model;

/**
 *
 * @author Micaela Jácome
 */
public class IBM extends Stock {
    private String symbol;
    private double price;

    public IBM(String symbol, double price) {
        this.symbol = symbol;
        this.price = price;
    }

    @Override
    public double getPrice() {
        return price;
    }

    public void setPrice(double price) {
        this.price = price;
        notifyObservers(Double.valueOf(price));
    }

    @Override
    public String getSymbol() {
        return symbol;
    }

    public void setSymbol(String symbol) {
        String oldSymbol = this.symbol;
        this.symbol = symbol;
        notifyObservers("Symbol changed from " + oldSymbol + " to " + symbol);
    }
}

